### PR TITLE
Add production terraform config

### PR DESCRIPTION
**Pull request by TimboTambo**
# Add terraform config for production
In preparation for release this week. 

Pretty much identical to the latest `non-prod` config, with the exception of the network resources (VNET/subnets/route table).

### Acceptance Criteria

- [ ] Running `terraform init` successfully initialises local state for production infrastructure
- [ ] Running `terraform plan` evaluates and displays resources to be created (should be all)

### Testing information

Follow acceptance criteria.

### Pre-flight Checklist

- [ ] Testing aligns with [testing strategy][testing strategy]
- [ ] DUXD review approval
- [ ] Accessibility Reviewed
- [ ] Product owner demo

### Pre-merge Checklist

- [ ] Branch test approved

[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"

### CODE REVIEW

File: infrastructure/environments/prod/main.tf [hunk 1]
```diff
@@ -80,5 +78,5 @@ module service_bus {
   environment         = var.ENVIRONMENT
   location            = var.REGION
   name                = local.service_bus_name
-  resource_group_name = azurerm_resource_group.products.name
+  resource_group_name = data.azurerm_resource_group.products.name
```
Comment (TimboTambo) [inline infrastructure/environments/prod/main.tf:81]: It'd be good to add the search service sku into the `<environment>/main.tf` files, setting it to `standard` for prod.

Comment (StuartHarris) [reply infrastructure/environments/prod/main.tf:82]: good idea. Done

File: .github/workflows/infrastructure.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Clone Repo
+        uses: actions/checkout@v2
+        with:
+          path: products
+
+      - name: Validate prod environment
```
Comment (StuartHarris) [inline .github/workflows/infrastructure.yaml:27]: nice!

**Comment by TimboTambo** [comment]
LGTM! @sgrowe you happy if this is merged?

**Comment by sgrowe** [comment]
> LGTM! @sgrowe you happy if this is merged?

Yeah, good with me 👍

**Pull request merged**